src/base/rss/rss_feed.cpp

```diff
@@ -207,13 +207,14 @@ void Feed::handleParsingFinished(const RSS::Private::ParsingResult &result)
         m_lastBuildDate = result.lastBuildDate;
 
         foreach (const QVariantHash &varHash, result.articles) {
-            auto article = Article::fromVariantHash(this, varHash);
-            if (article) {
+            try {
+                auto article = new Article(this, varHash);
                 if (!addArticle(article))
                     delete article;
                 else
                     m_dirty = true;
             }
+            catch (...) {}
```

Comment:
If you do not know what type of exceptions will be thrown by the article creation code, that code has to be redesigned. If you know, you have to catch those particular types.